php/libraries/Candidate.class.inc

```diff
@@ -606,8 +606,31 @@ class Candidate implements \LORIS\StudyEntities\AccessibleResource
 
         return $subprojList;
     }
+
     /**
-     * Returns first visit for a candidate based on Date_visit
+     * Returns the SubprojectID and subproject title of the candidate's last visit
+     * based on Date_visit
+     *
+     * @return array|null Array with values SubprojectID and title
+     *                    for the candidate's last visit or NULL if none exist
+     */
+    public function getSubprojectForMostRecentVisit(): ?array
+    {
+        $factory = NDB_Factory::singleton();
+        $db      = $factory->database();
+
+        $candID = $this->getCandID();
+        $query  = "SELECT SubprojectID, title FROM session
+          LEFT JOIN subproject USING (SubprojectID)
+          WHERE CandID=:cid AND Date_visit IS NOT NULL
+          ORDER BY Date_visit DESC LIMIT 1";
+        $where  = array('cid' => $candID);
+        $result = $db->pselect($query, $where);
+        return $result ?? null;
```

Comment:
I think Travis is having an issue with this part. `pselect()` returns the type array so I believe that if this query has no result, the function will return an empty array (instead of null). Your unit tests are doing assertions on `''`  which causes an error because the actual returned value is `array()`
```suggestion
        return $result;
```